lib/loader.js

```diff
@@ -61,38 +82,38 @@ const readOrError = async (file, options = {}) => {
     }
 }
 
-function deepMergeRules(ruleNickname, skipRules, rules = []) {
-    // TODO Eventually support more than just core files
-    const ruleFile = path.join(__dirname, '../rules/' + ruleNickname + '.json');
-    const content = fs.readFileSync(ruleFile, 'utf8');
-    const data = yaml.safeLoad(content, { json: true });
+const recursivelyLoadRuleFiles = async (file, loadedFiles, options) => {
+    const { verbose } = options;
+    if (verbose > 1) console.log('GET ' + file);
 
-    if (typeof data.require == 'string') {
-        rules = deepMergeRules(data.require, rules);
+    let data;
+    if (file && file.startsWith('http')) {
+        data = await fetchUrl(file);
+    }
+    else {
+        const ruleFile = path.join(__dirname, '../rules/' + file + '.json');
+        const content = fs.readFileSync(ruleFile, 'utf8');
+        data = yaml.safeLoad(content, { json: true });
     }
 
-    for (const r in data.rules) {
-        const rule = data.rules[r];
-        if (!rule.enabled) continue;
-        if (skipRules.indexOf(rule.name) !== -1) continue;
-        if (!Array.isArray(rule.object)) rule.object = [rule.object];
-        if (rule.alphabetical && rule.alphabetical.properties && !Array.isArray(rule.alphabetical.properties)) {
-            rule.alphabetical.properties = [rule.alphabetical.properties];
-        }
-        if (rule.truthy && !Array.isArray(rule.truthy)) rule.truthy = [rule.truthy];
-        rules.push(rule);
+    loadedFiles.push(file);
+
+    if (typeof data.require == 'string') {
+        loadedFiles.concat(recursivelyLoadRuleFiles(data.require, loadedFiles, options));
     }
 
-    return rules;
+    // Tell the linter about these new rules
+    linter.createNewRules(data.rules);
+
+    return loadedFiles;
 }
 
-const loadRules = (loadFiles, skipRules = []) => {
+const loadRuleFiles = async (loadFiles, options = {}) => {
+    const { skip = [], verbose } = options;
+    linter.initialize({ skip });
     const files = (loadFiles.length > 0 ? loadFiles : ['default']);
-    let loadedRules = [];
-    for (const f in files) {
-        loadedRules = loadedRules.concat(deepMergeRules(files[f], skipRules));
-    }
-    return loadedRules;
+    const promises = files.map(file => recursivelyLoadRuleFiles(file, [], { verbose }));
```

Comment:
Gah ok so loading URLs does not actually work it seems. It loads the file, but i dont think the promises ever get resolved. In my debugging I added these console lines:

```
    console.log('Got some promises', promises);
    const foo = await Promise.all(promises);
    console.log('Did they resolve?', foo);
```

The output with --verbose is...

```
GET https://gist.githubusercontent.com/philsturgeon/b5295d402f2dc94b86aa32eddb1fa100/raw/1f1c3516294c8c8ebe294aea8b56c7116bce28c0/rules.json
Got some promises [ Promise { <pending> } ]
Specification is valid, with 0 lint errors
```

So it seems to junk out somewhere in the Promise.all... What the heck is that about?

Reply:
The death is happening mid fetchUrl, this code outputs `got here https://gist.githubusercontent.com/philsturgeon/b5295d402f2dc94b86aa32eddb1fa100/raw/06eec564a2f8c9b972930981c165ebd7b00b8322/rules.json` but never outputs the data or an error...

```

const fetchUrl = async (url) => {
    console.log('got here', url);

    fetch(url).then(data => {
        console.log('data', data)
    }, err => {
        console.log('err', err)
    });

```

Reply:
You may be missing some wrapping try/catch mechanisms in your code.